src/psyclone/nemo.py

```diff
@@ -261,20 +141,18 @@ def inline(self, _):
     @property
     def gen(self):
         '''
-        Generate the (updated) fparser2 AST for the NEMO code represented
-        by this NemoPSy object.
+        Generate the Fortran for the NEMO code represented by this
+        NemoPSy object.
 
-        :returns: the fparser2 AST for the Fortran code.
-        :rtype: :py:class:`fparser.two.Fortran2003.Main_Program` or \
-                :py:class:`fparser.two.Fortran2003.Subroutine_Subprogram` or \
-                :py:class:`fparser.two.Fortran2003.Function_Subprogram`.
-        '''
-        # Walk down our Schedule and update the underlying fparser2 AST
-        # to account for any transformations
-        self.invokes.update()
+        :returns: the Fortran code.
+        :rtype: str
 
-        # Return the fparser2 AST
-        return self._ast
+        '''
+        from psyclone.psyir.backend.fortran import FortranWriter
+        fwriter = FortranWriter()
+        # Ensure any domain-specific concepts are lowered
+        self._container.lower_to_language_level()
+        return fwriter(self._container)
 
 
 class NemoInvokeSchedule(InvokeSchedule):
```

Comment:
Oh yes, done.

Reply:
In fact, it turns out that if I do that then we get the wrong name displayed in the schedule because a number of the parent classes do set `_text_name`. It was beginning to spiral so I've put back `_text_name` in this class for now.